src/api/app/models/cloud/upload_job.rb

```diff
@@ -39,6 +41,16 @@ def target_validator_class
       @target_validator_class ||= "::Cloud::Params::#{target.capitalize}".constantize
     end
 
+    def to_xml(options = {})
+      builder = Builder::XmlMarkup.new(options)
+      builder.cloud_upload_job(id: id) do |xml|
+        xml.target target
+        xml.filename filename
+        xml.vpc_subnet_id vpc_subnet_id
```

Comment:
Why is the vpc_subnet_id in the upload job and not in the params?

Reply:
I've removed the ` to_xml ` methods again. All actions now directly return the backend data.